### ISSUE TITLE
No entities found in markdown training file

### ISSUE THREAD

**Issue by JLMoszkowicz** [RasaHQ/rasa_nlu#1360]
**Rasa NLU version**: 0.12.3

**Operating system** (windows, osx, ...): Linux

**Content of model configuration file**:
```yml
language: en

pipeline: spacy_sklearn
```

**Issue**: We are not able to find the many distinct entities we have in our training data, which is in markdown format, under this version.  It was working previously when we were running version 0.10.5 of Rasa NLU, but now we only find the intents.  Our training data includes comments in the markdown, if that is relevant.  Any ideas?

**Comment by akelad** [comment]
I'm not sure I understand the question. What do you mean by "find" entities/intents? I assume you wrote the markdown file so it's up to you to label them... 
If you could post your training data file, that might help me understand what's going wrong

**Comment by JLMoszkowicz** [comment]
So, our training file is full of labeled entities in the intent examples using square brackets around the entity example and parentheses around the label. The problem seems to be with training a model using the training file. We've tried using both spacy and tensorflow. Both recognize the correct number of intents from the training file during the model creation, but they report finding zero distinct entities. I assume that means we're failing to train our model on our entities.

**Comment by akelad** [comment]
Please post your training data file, and the training logs. I'd guess they're labeled incorrectly.

**Comment by JLMoszkowicz** [comment]
@akelad : Here is a small excerpt from our our training data.  Do you see anything that would break rasa_nlu.train?  Where would I find training logs?

```
## intent:Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rghvRspgdSUOsHFZDonGKfg <!-- (#$RASATrainingTemplateFn #$EMAApplicationKBQ-WhatConditionsOfTypeAreOfInterestToThoseAffectedByAilmentType) -->
- which [cardiovascular](Mx8Ngx4rzV0J3b91SUqszjxHn7teiR4rwQw4WZwpEbGdrcN5Y29ycA-kdGhlIHR5cGUgb2YgY29tcGxpY2F0aW9uIGFza2VkIGFib3V0) conditions are important for someone who has [diabetes](Mx8Ngx4rzV0J3b91SUqszjxHn7teiR4rCgf3HsC_SpGVospPLCIYBw-cdGhlIGFpbG1lbnQgdHlwZSBhc2tlZCBhYm91dA) to know about <!-- #<(#$TheNamedFn #$PhysiologicalCondition "the type of complication asked about")> #<(#$TheNamedFn #$AilmentConditionType "the ailment type asked about")> -->
- which [cardiovascular](Mx8Ngx4rzV0J3b91SUqszjxHn7teiR4rwQw4WZwpEbGdrcN5Y29ycA-kdGhlIHR5cGUgb2YgY29tcGxpY2F0aW9uIGFza2VkIGFib3V0) conditions are important for someone who has [diabetes](Mx8Ngx4rzV0J3b91SUqszjxHn7teiR4rCgf3HsC_SpGVospPLCIYBw-cdGhlIGFpbG1lbnQgdHlwZSBhc2tlZCBhYm91dA) <!-- #<(#$TheNamedFn #$PhysiologicalCondition "the type of complication asked about")> #<(#$TheNamedFn #$AilmentConditionType "the ailment type asked about")> -->
- which [cardiovascular](Mx8Ngx4rzV0J3b91SUqszjxHn7teiR4rwQw4WZwpEbGdrcN5Y29ycA-kdGhlIHR5cGUgb2YgY29tcGxpY2F0aW9uIGFza2VkIGFib3V0) conditions are important for someone who has [diabetes treated with insulin](Mx8Ngx4rzV0J3b91SUqszjxHn7teiR4rCgf3HsC_SpGVospPLCIYBw-cdGhlIGFpbG1lbnQgdHlwZSBhc2tlZCBhYm91dA) to know about <!-- #<(#$TheNamedFn #$PhysiologicalCondition "the type of complication asked about")> #<(#$TheNamedFn #$AilmentConditionType "the ailment type asked about")> -->
- which [cardiovascular](Mx8Ngx4rzV0J3b91SUqszjxHn7teiR4rwQw4WZwpEbGdrcN5Y29ycA-kdGhlIHR5cGUgb2YgY29tcGxpY2F0aW9uIGFza2VkIGFib3V0) conditions are important for someone who has [diabetes treated with insulin](Mx8Ngx4rzV0J3b91SUqszjxHn7teiR4rCgf3HsC_SpGVospPLCIYBw-cdGhlIGFpbG1lbnQgdHlwZSBhc2tlZCBhYm91dA) <!-- #<(#$TheNamedFn #$PhysiologicalCondition "the type of complication asked about")> #<(#$TheNamedFn #$AilmentConditionType "the ailment type asked about")> -->
```

**Comment by akelad** [comment]
What's the reason for these really long entity names and intent names? Seems very odd and probably not very useful when trying to understand what the model has detected. This strange format is most likely what's causing the issues. 
And the training logs are the logs that get printed during the training of the model, you said that they said no entities were found?

**Comment by JLMoszkowicz** [comment]
The crazy long names are IDs into an internal system.  I know they look terrible, but we were getting normal results when we were using 0.10.5 of Rasa NLU, so I don't think they're the problem, unless there's been a change to the markdown format for the training data.

Here is the training log.  The training script was able to identify 33 distinct intents, even with the long intent names, but no entities were found.

```
python -m rasa_nlu.train -c config_spacy.yml --data trainingData/ema-training-9418-1121.md -o models --project ema --verbose
2018-09-04 11:24:31 INFO     rasa_nlu.utils.spacy_utils  - Trying to load spacy model with name 'en'
2018-09-04 11:24:33 INFO     rasa_nlu.components  - Added 'nlp_spacy' to component cache. Key 'nlp_spacy-en'.
2018-09-04 11:24:33 INFO     rasa_nlu.training_data.loading  - Training data format of trainingData/ema-training-9418-1121.md is md
2018-09-04 11:24:34 INFO     rasa_nlu.training_data.training_data  - Training data stats: 
	- intent examples: 50545 (33 distinct intents)
	- Found intents: 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rco48Hi5_Qo-0SVUUdHLt0g', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rOnAZDgzISSi7ImUgC9z8cw', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rUHCBowjAQkuQFN4B3OgBWQ', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rWdwfZ8qHQM-v2jmV2-j_Xw', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4r0mxkuW8OTc2FdEirC3w3Lw', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4riVmMZMLAQpWoJxWPublgDA', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rjhgNe45hShSpszI-YvClfg', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4r0lMn1o0SRfGmuvV5WH2Bhg', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rcTEUkghrRsGz_2_OBr2pGA', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rGtOlA4vUSaStBwMLsPIJSA', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rESfq38DKSv6AivRDbCWHaw', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rUYGTi0DbSOiZ6-Kl8xjw8g', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4ry7aKwWPVSKuN3U_lIgABCw', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rbhvCrhP1SVqDaGGytvjw_A', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rghvRspgdSUOsHFZDonGKfg', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rh18ohb3XR262FBzZUKAykg', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rut2jynB2TjONioE0otxeag', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rGYYRRMt3TUuzcBTPrKIPtQ', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4r6-mlw-UYRp23tRGSg1pXaw', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rm6DfBRxdRSiZtjYYkxRPgA', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rqZcall6PQ1GqqbTwOVe1RA', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rG7O-bTaESN22jalW3Rotqw', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rl5QYRhlTTXK5KS4uUE72JQ', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4r8oEsGQpMRsm4OoBPLxSLxA', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rsDAUmHMFRRGqx4hvOxsMsQ', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rOZFbIu2qQZSVytpf0L0I7A', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rPghiJMh5SOuekpf7_xEYeg', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rsFzXckq4RXWnVhK6IJZJ9A', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rD0JB6Z1WSi6_kVrjJl6H5A', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rFU43pHOPQx-oP3avQyhMmg', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rxahamvInTW6amThOl_Sl-A', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4rI_DY5j8MTFaAYTV33II19g', 'Mx8Ngh4rvb-Vd4uJEeiDCkhNfs2N8x4r2_cFKA_ZR0iPdPLA2bHTwQ'
	- entity examples: 0 (0 distinct entities)
	- found entities: 

2018-09-04 11:24:36 INFO     rasa_nlu.model  - Starting to train component nlp_spacy
2018-09-04 11:34:41 INFO     rasa_nlu.model  - Finished training component.
2018-09-04 11:34:41 INFO     rasa_nlu.model  - Starting to train component tokenizer_spacy
2018-09-04 11:34:45 INFO     rasa_nlu.model  - Finished training component.
2018-09-04 11:34:45 INFO     rasa_nlu.model  - Starting to train component intent_featurizer_spacy
2018-09-04 11:34:46 INFO     rasa_nlu.model  - Finished training component.
2018-09-04 11:34:46 INFO     rasa_nlu.model  - Starting to train component intent_entity_featurizer_regex
2018-09-04 11:34:46 INFO     rasa_nlu.model  - Finished training component.
2018-09-04 11:34:46 INFO     rasa_nlu.model  - Starting to train component ner_crf
2018-09-04 11:34:46 INFO     rasa_nlu.model  - Finished training component.
2018-09-04 11:34:46 INFO     rasa_nlu.model  - Starting to train component ner_synonyms
2018-09-04 11:34:46 INFO     rasa_nlu.model  - Finished training component.
2018-09-04 11:34:46 INFO     rasa_nlu.model  - Starting to train component intent_classifier_sklearn
Fitting 2 folds for each of 6 candidates, totalling 12 fits
/usr/local/lib/python2.7/dist-packages/sklearn/metrics/classification.py:1135: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
[Parallel(n_jobs=1)]: Done  12 out of  12 | elapsed: 14.0min finished
2018-09-04 11:50:39 INFO     rasa_nlu.model  - Finished training component.
2018-09-04 11:50:40 INFO     rasa_nlu.model  - Successfully saved model into '/home/jmoszko/rasaWorkingDirectory/models/ema/model_20180904-115039'
2018-09-04 11:50:40 INFO     __main__  - Finished training

```

**Comment by akelad** [comment]
it seems the dashes in the entity names are causing the problems

**Comment by JLMoszkowicz** [comment]
Interestingly, we were able to get things working again by switching to json for our training data. So, we have a workaround, but it's curious that the dashes didn't seem to be an issue under version 0.10.5. I noticed there have been some features added in recent versions to what you can do with markdown. Perhaps one of them introduced a new bug?

**Comment by akelad** [comment]
That is strange indeed, @tmbo any ideas what might have changed to cause this?

**Comment by tmbo** [comment]
Yes that is an issue - I created a PR to fix the regex that will match the entities and created a couple tests to ensure it stays that way.

**Comment by JLMoszkowicz** [comment]
Thanks for all of your quick responses!
